src/OILibraryImpl.cpp

```diff
@@ -152,114 +144,53 @@ int OILibraryImpl::compileCode() {
   auto objectPath =
       fs::path((boost::format("/dev/fd/%1%") % objectMemfd).str());
 
-  if (_self->opts.forceJIT) {
-    struct drgn_program *prog = symbols->getDrgnProgram();
-    if (!prog) {
-      return Response::OIL_COMPILATION_FAILURE;
-    }
-    struct drgn_symbol *sym;
-    if (auto err = drgn_program_find_symbol_by_address(
-            prog, (uintptr_t)_TemplateFunc, &sym)) {
-      LOG(ERROR) << "Error when finding symbol by address " << err->code << " "
-                 << err->message;
-      drgn_error_destroy(err);
-      return Response::OIL_COMPILATION_FAILURE;
-    }
-    const char *name = drgn_symbol_name(sym);
-    drgn_symbol_destroy(sym);
+  struct drgn_program *prog = symbols->getDrgnProgram();
+  if (!prog) {
+    return Response::OIL_COMPILATION_FAILURE;
+  }
+  struct drgn_symbol *sym;
+  if (auto err = drgn_program_find_symbol_by_address(
+          prog, (uintptr_t)_TemplateFunc, &sym)) {
+    LOG(ERROR) << "Error when finding symbol by address " << err->code << " "
+               << err->message;
+    drgn_error_destroy(err);
+    return Response::OIL_COMPILATION_FAILURE;
+  }
+  const char *name = drgn_symbol_name(sym);
+  drgn_symbol_destroy(sym);
 
-    auto rootType = symbols->getRootType(irequest{"entry", name, "arg0"});
-    if (!rootType.has_value()) {
-      LOG(ERROR) << "Failed to get type of probe argument";
-      return Response::OIL_COMPILATION_FAILURE;
-    }
+  auto rootType = symbols->getRootType(irequest{"entry", name, "arg0"});
```

Comment:
Add a comment about changing this to the new `drgn_symbol -> drgn_type` pipeline.